sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpAnnotatedMessage.java

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.models;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Amqp representation of the message.
+ */
+public final class AmqpAnnotatedMessage {
+    private final AmqpMessageBody amqpMessageBody;
+    private Map<String, Object> applicationProperties;
+    private Map<String, Object> deliveryAnnotations;
+    private Map<String, Object> messageAnnotations;
+    private Map<String, Object> footer;
+    private AmqpMessageHeader header;
+    private AmqpMessageProperties properties;
+
+    /**
+     * Creates instance of {@link AmqpAnnotatedMessage} with given {@link AmqpMessageBody}.
+     *
+     * @param body to be set on amqp message.
+     */
+    public AmqpAnnotatedMessage(AmqpMessageBody body) {
+        this.amqpMessageBody = Objects.requireNonNull(body, "'body' cannot be null.");
+    }
+
+    /**
+     * Creates instance of {@link AmqpAnnotatedMessage} with given {@link AmqpAnnotatedMessage}.
+     *
+     * @param message used to create another instance of {@link AmqpAnnotatedMessage}.
+     */
+    public AmqpAnnotatedMessage(AmqpAnnotatedMessage message) {
+        Objects.requireNonNull(message, "'message' cannot be null.");
+        this.amqpMessageBody = Objects.requireNonNull(message.getBody(), "'message.body' cannot be null.");
+        this.applicationProperties = message.getApplicationProperties();
+        this.deliveryAnnotations = message.getDeliveryAnnotations();
+        this.messageAnnotations = message.getMessageAnnotations();
+        this.header = message.getHeader();
+        this.properties = message.getProperties();
+    }
+
+    /**
+     * Gets the {@link Map} of application properties.
+     * @return The application properties.
+     */
+    public Map<String, Object> getApplicationProperties() {
+        if (this.applicationProperties == null) {
+            this.applicationProperties = new HashMap<>();
+        }
+        return applicationProperties;
+    }
+
+    /**
+     * Gets the {@link AmqpMessageBody}.
+     *
+     * @return the {@link AmqpMessageBody} object.
+     */
+    public AmqpMessageBody getBody() {
+        return amqpMessageBody;
+    }
+
+    /**
+     * Gets the {@link Map} representation of delivery annotations.
+     *
+     * @return the {@link Map} representation of delivery annotations.
+     */
+    public Map<String, Object> getDeliveryAnnotations() {
+        if (deliveryAnnotations == null) {
+            this.deliveryAnnotations = new HashMap<>();
```

Comment:
Inconsistent use of `this`. Why do the instantiation here vs in the constructor and make it final. We always do .getDeliveryAnnotations() when serialising or deserialising, so weren't not saving any time.

Reply:
I took motivation from Amqp library implementation but ,you are right we are not saving any time. So I will put it back in Contructor.